components/app/explore/DatasetWidget.js

```diff
@@ -112,14 +80,52 @@ class DatasetWidget extends React.Component {
     );
   }
 
+  getWidgetOrLayer() {
+    if (this.state.hasWidget) {
+      return {
+        ...this.state.widget.attributes,
+        ...{ id: this.state.widget.id }
+      };
+    }
+    if (this.state.hasLayer) {
+      return {
+        ...this.state.layer.attributes,
+        ...{ id: this.state.layer.id }
```

Comment:
Not my code, by of course, this makes sense. When I fixed the eslint issues I didn't change the logic.